lib/install.go

```diff
@@ -138,97 +136,78 @@ func install(product Product, tfversion string, binPath string, installPath stri
 	/* set symlink to desired version */
 	CreateSymlink(installFileVersionPath, binPath)
 	logger.Infof("Switched terraform to version %q", tfversion)
-	addRecent(product, tfversion, installPath) //add to recent file for faster lookup
+	addRecentVersion(product, tfversion, installPath) //add to recent file for faster lookup
 	return
 }
 
-// addRecent : add to recent file
-func addRecent(product Product, requestedVersion string, installPath string) {
+// getRecentFileName : Obtain recent version filename for product
+func getRecentFileName(product Product) string {
+	return recentFilePrefix + product.GetId() + ".json"
+}
 
+// addRecent : add to recent file
+func addRecentVersion(product Product, requestedVersion string, installPath string) {
 	installLocation = GetInstallLocation(installPath) //get installation location -  this is where we will put our terraform binary file
 	recentFilePath := filepath.Join(installLocation, getRecentFileName(product))
 
-	fileExist := CheckFileExist(recentFilePath)
-	if fileExist {
-		lines, errRead := ReadLines(recentFilePath)
+	// Obtain pre-existing latest version
+	versions := getRecentVersions(product, installPath)
 
-		if errRead != nil {
-			logger.Errorf("Error reading %q file: %v", recentFilePath, errRead)
-			return
-		}
-
-		for _, line := range lines {
-			if !validVersionFormat(line) {
-				logger.Infof("File %q is dirty (recreating cache file)", recentFilePath)
-				RemoveFiles(recentFilePath)
-				CreateRecentFile(requestedVersion, installPath, recentFilePath)
-				return
-			}
+	// Check for requestedVersion in versions list
+	for versionIndex, versionVal := range versions {
+		if versionVal == requestedVersion {
+			versions = append(versions[:versionIndex], versions[versionIndex+1:]...)
 		}
+	}
 
-		versionExist := versionExist(requestedVersion, lines)
+	// Add new version to start of slice
+	versions = append([]string{requestedVersion}, versions...)
+	if len(versions) > 3 {
+		versions = versions[0:2]
+	}
```

Comment:
This pushed me to think that we may allow user to customize number of recent versions we keep for them. WDYT?

Reply:
I'm planning to merge this PR (which will go into #435) and I'll re-raise this comment there :)

Reply:
Thank you

Reply:
Instead of letting users customize, lets leave it at 5 for now (increase from 3). I don't think we should over-engineer this.